### DISCUSSION TITLE
Best practise to chain API calls?

### DISCUSSION THREAD

**Discussion by CoenraadS** [usebruno/bruno#942] [Q&A] [answered]
Hello

I was wondering what is the best practice to implement something like

1. GET /api/foo/latest => returns `id`

2. GET API/foo/{id} 

Either automatically running after each other or manually, both are fine for me.

Would setting the ID in environment variable be the way to go? I tried to also understand `vars` https://docs.usebruno.com/scripting/vars.html, but didn't really understand if they could be used across APIs

**Comment by Sl-Alex** [comment, marked as answer]
I'd do it like this:

Request 1. Execute GET /api/foo/latest, get the ```ìd``` out of the body and put it to the collection variable in the post-request script:
```js
bru.setVar('app-foo-latest-id', res.body.id);
```
Request 2. Execute GET /api/foo/ request with the following pre-request script:
```js
req.setUrl(req.getUrl() + bru.getVar('app-foo-latest-id'));
```
Collection variables work across different requests in the same collection.

**Reply by CoenraadS** [reply]
Thanks, I just checked and I could also use `{{app-foo-latest-id}}` in the URL and it would also work with vars. Great!

**Reply by carlinmack** [reply]
Now instead of 

```bru
script:post-response {
  bru.setVar('record_id', res.body.id);
}
```

you can do 

```bru
vars:post-response {
  record_id: res.body.id
}
```

**Comment by svenjacobs** [comment]
Hello all,

is there a solution to automatically chain two or more calls together? Let's say request `A` requires an authentication token which request `B` retrieves. How can I always call request `B` before `A`? I know that I can put both requests in a folder and execute the folder but that would mean that I have to duplicate request `B` into each folder if I have multiple requests that require a token.

Any ideas?

**Reply by TarekSalha** [reply]
if its just auth, you can put the token acquisition directly into the collection. then, every call gets the token first and executes afterwards

**Reply by svenjacobs** [reply]
> if its just auth, you can put the token acquisition directly into the collection. then, every call gets the token first and executes afterwards

Yes, this is what I eventually went for. I'm not sure if this was available when I wrote this question.
